File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.java
class Solution {
    public int subsetXORSum(int[] nums) {
        int n = nums.length, total = 0;
        for (int mask = 0; mask < (1 << n); mask++) {
            int subsetXor = 0;
            for (int i = 0; i < n; i++) {
                if ((mask & (1 << i)) != 0) {
                    subsetXor ^= nums[i];
                }
            }
            total += subsetXor;
        }
        return total;
    }
}

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    static int subsetXORSum(vector<int>& nums, int i=0, int xorsum=0) {
        return (i==nums.size())?xorsum:subsetXORSum(nums, i+1, xorsum^nums[i])+subsetXORSum(nums, i+1, xorsum);      
    }
};